src/ext/default.c

```diff
@@ -27,55 +28,82 @@
 // normalize TF by number of tokens (weighted)
 #define NORM_DOCLEN 2
 
-#define EXPLAIN(explain, indexResult, fmt, args...)                                            \
-        if (explain) {                                                                \
-          char *tempStr = indexResult->scoreExplainStr;                               \
-          rm_asprintf((char ** restrict)&indexResult->scoreExplainStr, fmt, ##args);     \
-          rm_free(tempStr);                                                              \
-        }                                                                             \
+static inline void EXPLAIN(RSScoreExplain *scrExp, char *fmt, ...) {
```

Comment:
changhe casing to lower or cmel case. no need for inline, eitjer. also, you can wrap in macro eg `define EXPLAIN(e, ...) if (e) {explain(e, ##__vaargs__)`

Reply:
@mnunberg  :+1: